### PR TITLE
[CALCITE-827] Add a LogicalProject on top of LogicalWindow to permute…

### PR DESCRIPTION
**Pull request by hsuanyi**
… the columns in the same order as specified in the SELECT list

### CODE REVIEW

**Comment by hsuanyi** [comment]
Had addressed amansinha100's comment

**Comment by julianhyde** [comment]
It looks good -- reviewing further now.

Can you also add a SQL test case for this issue to winagg.oq? I think it would be straightforward and reduce the chance of regression.

File: core/src/main/java/org/apache/calcite/rel/logical/LogicalWindow.java [hunk 1]
```diff
@@ -116,11 +116,13 @@ public static RelNode create(
     // Build a list of groups, partitions, and aggregate functions. Each
     // aggregate function will add its arguments as outputs of the input
     // program.
+    final Map<RexOver, RexOver> origToNewOver = new HashMap<>();
```
Comment (julianhyde) [inline core/src/main/java/org/apache/calcite/rel/logical/LogicalWindow.java]: Use IdentityHashMap - same effect, less confusing

**Comment by hsuanyi** [comment]
Add test case in winagg.oq; Julian, can you help review?

**Comment by julianhyde** [comment]
I've merged it to my https://github.com/julianhyde/incubator-calcite/tree/new-master branch. The pull request will get closed when I merge to apache master, after the release. Thanks for the pull request!

**Pull request closed without merging**